ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ipv6_neighbors.schema.yml

```
@@ -0,0 +1,55 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ipv6_neighbor:
+    type: dict
+    keys:
+      static_entries:
+        type: list
+        description: Static IPv6 neighbor entries.
+        items:
+          type: dict
+          keys:
+            ipv6_address:
+              type: str
+              required: true
+              description: IPv6 address of neighbor.
+            vrf:
+              type: str
+            interface:
+              type: str
+              required: true
+              description: |-
+                Interface name.
+                Example - Ethernet4
+                          Loopback4-6
+                          Port-channel4,7
+                          Management 1
+                          Tunnel10
+                          Vlan101,102
+            mac_address:
+              type: str
+              required: true
+              description: |-
+                Hardware address of neighbor.
+                Example - aaaf.1234.bcbf
+              pattern: "^(?:[0-9A-Fa-f]{1,4}\\.){2}[0-9A-Fa-f]{1,4}$"
```

The mac addresses *must* have four characters, so this pattern is not valid.
```suggestion
              pattern: "^([0-9A-Fa-f]{4}\\.){2}[0-9A-Fa-f]{4}$"
```

---

But CLI accept it if we dont give the four char, then EOS convert it at the backend.
```
wan1-dc1(config)#ipv6 neighbor 11:2:33:44:55:66:77:88 ethernet 1 A.B.C ?
  <cr>  

wan1-dc1(config)#ipv6 neighbor 11:2:33:44:55:66:77:88 ethernet 1 A.B.C 
wan1-dc1(config)#
```

```
wan1-dc1(config)#show running-config section ipv6
ipv6 neighbor 11:2:33:44:55:66:77:88 Ethernet1 00:0a:00:0b:00:0c
wan1-dc1(config)#
```

---

As for the other comment, we model running config. Not EOS input.

---

running-config has the format like aa:bb:cc:dd:ee:ff

would be good to use that format as this is also used in many other places

---

updated